modules/couchbase/src/main/java/org/testcontainers/couchbase/CouchbaseWaitStrategy.java

```diff
@@ -0,0 +1,164 @@
+/*
+ * Copyright (c) 2016 Couchbase, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.testcontainers.couchbase;
+
+import com.couchbase.client.deps.com.fasterxml.jackson.databind.JsonNode;
+import com.couchbase.client.deps.com.fasterxml.jackson.databind.ObjectMapper;
+import com.google.common.base.Strings;
+import com.google.common.io.BaseEncoding;
+import lombok.Cleanup;
+import lombok.extern.slf4j.Slf4j;
+import org.rnorth.ducttape.TimeoutException;
+import org.testcontainers.containers.ContainerLaunchException;
+import org.testcontainers.containers.wait.strategy.AbstractWaitStrategy;
+
+import java.io.IOException;
+import java.net.HttpURLConnection;
+import java.net.URI;
+import java.net.URL;
+import java.util.Optional;
+import java.util.concurrent.TimeUnit;
+
+import static org.rnorth.ducttape.unreliables.Unreliables.retryUntilSuccess;
+
+/**
+ * @author ldoguin
+ * created on 18/07/16.
+ */
+@Slf4j
+public class CouchbaseWaitStrategy extends AbstractWaitStrategy {
```

Comment:
I have a feeling that this class can be removed or dramatically simplified by using `HttpWaitStrategy` + `responsePredicate` :)

Reply:
Yes you are absolutely right, I'm doing it now!